File: en Python/exercice4_1.py
import os
p1_r, p1_w = os.pipe()
p2_r, p2_w = os.pipe()

pid = os.fork()
if pid == 0:
    os.close(p1_r)
    os.close(p2_w)
    mot = input("entrez un mot: ")
    os.write(p1_w, mot.encode())

    resultat = os.read(p2_r, 10).decode()
    if resultat == "1": print("Le mot est dans le fichier")
    else: print("Le mot n est pas dans le fichier")
    os._exit(0)
else:
    os.close(p1_w)
    os.close(p2_r)
    mot = os.read(p1_r, 255).decode().strip()
    trouve = 0
    with open("fic.txt", "r") as f:
        for line in f:
            if mot in line.split():
                trouve = 1
                break
    os.write(p2_w, str(trouve).encode())
    os.wait()

File: En C/exercice4_1.c
#include <stdio.h>
#include <unistd.h>
#include <string.h>
#include <sys/wait.h>
int main()
{
    int p1[2], p2[2]; 
    pipe(p1); 
    pipe(p2); 
    pid_t pid = fork();

    if (pid == 0)
    {
        char mot[255];
        int resultat; 
        close(p1[0]);
        close(p2[1]); 
        printf("entrez un mot: ");
        scanf("%s", mot);
        write(p1[1], mot, 255);
        read(p2[0], &resultat,sizeof(int));
        if (resultat == 1)printf("Le mot est dans le fichier\n") ;
        else printf ("Le mot n est pas dans le fichier\n") ;
    }
    else 
    {
        FILE *pf;
        char mot[255], motFic[255];
        int r = 0;
        close(p1[1]); 
        close(p2[0]); 
        read(p1[0], mot, 255);
        pf = fopen("fic.txt", "r");
        while (fscanf(pf, "%s", motFic) == 1)
            if(strcmp (mot , motFic) == 0) { r = 1 ; break ;}
        fclose(pf);
        write(p2[1], &r, sizeof(int));
        wait(NULL);
    }
    return 0;
}